src/app/shared/dialogs/change-password.directive.ts

```diff
@@ -11,32 +11,33 @@ import { CustomValidators } from '../../validators/custom-validators';
 import { ValidatorService } from '../../validators/validator.service';
 import { StateService } from '../state.service';
 
-const changePasswordFields = [
-  {
-    'label': 'Old Password',
-    'type': 'textbox',
-    'inputType': 'password',
-    'name': 'oldPassword',
-    'placeholder': 'Old Password',
-    'required': true
-  },
-  {
-    'label': 'Password',
-    'type': 'textbox',
-    'inputType': 'password',
-    'name': 'password',
-    'placeholder': 'Password',
-    'required': true
-  },
-  {
-    'label': 'Confirm Password',
-    'type': 'textbox',
-    'inputType': 'password',
-    'name': 'confirmPassword',
-    'placeholder': 'Confirm Password',
-    'required': true
-  }
-];
+function getFormFields() {
```

Comment:
@singharpita This doesn't need to be changed to a function.  Could you revert this back to a variable?

Reply:
@paulbert dialog is updating the variable on its close with last state change making visibility of password field to be visible when we open change password again. So either we will have to move const into opendialog function or we need to reinitialize variable. That's why it is converted to function to reset the variable

Reply:
That's because you are modifying the `field` object directly in `DialogsFormComponent` when toggling the password view.  I don't think that's a great idea, this object should not be modified, properties on the component should be modified instead.